### PR TITLE
media-sound/supercollider: sign up for proxy-maint and bump to 3.11.0

### PR DESCRIPTION
**Pull request by marcan**

### CODE REVIEW

File: media-sound/supercollider/supercollider-3.11.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+DESCRIPTION="An environment and a programming language for real time audio synthesis."
+HOMEPAGE="https://supercollider.github.io/"
+SRC_URI="https://github.com/supercollider/supercollider/releases/download/Version-${PV}/SuperCollider-${PV}-Source.tar.bz2"
+
+LICENSE="GPL-2 gpl3? ( GPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="cpu_flags_x86_sse cpu_flags_x86_sse2 debug emacs +fftw gedit +gpl3 jack qt5 server +sndfile static-libs vim X zeroconf"
+
+REQUIRED_USE="qt5? ( X )"
```
Comment (a17r) [inline media-sound/supercollider/supercollider-3.11.0.ebuild:17]: is that still accurate? quite often this is accompanied by a dependency on qtx11extras but at least from the ebuild that does not seem to be the case.

File: media-sound/supercollider/supercollider-3.11.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	)
+	zeroconf? ( net-dns/avahi )
+"
+DEPEND="${RDEPEND}
+	dev-libs/icu
+	emacs? ( >=app-editors/emacs-23.1:* )
+	gedit? ( app-editors/gedit )
+	qt5? (
+		dev-qt/qtconcurrent:5
+		dev-qt/qtdeclarative:5
```
Comment (a17r) [inline media-sound/supercollider/supercollider-3.11.0.ebuild:59]: that looks odd. no qml files installed or linking to qtdeclarative?

File: media-sound/supercollider/supercollider-3.11.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	server? ( !app-admin/supernova )
+	sndfile? ( media-libs/libsndfile )
+	X? (
+		x11-libs/libX11
+		x11-libs/libXt
+	)
+	zeroconf? ( net-dns/avahi )
+"
+DEPEND="${RDEPEND}
+	dev-libs/icu
```
Comment (a17r) [inline media-sound/supercollider/supercollider-3.11.0.ebuild:54]: no linking to icu? that would be a rare sight.

File: media-sound/supercollider/supercollider-3.11.0.ebuild (hunk 1 above)
Comment (marcan) [reply media-sound/supercollider/supercollider-3.11.0.ebuild:17]: Upstream definitely has X as a dependency of the Qt GUI in their buildsystem (for Linux), and a pile of codepaths referencing Q_WS_X11 there (and not a trace of wayland). There is also optional X support outside of the GUI.

I'm not confident we can do much about this here.

File: media-sound/supercollider/supercollider-3.11.0.ebuild (hunk 3 above)
Comment (marcan) [reply media-sound/supercollider/supercollider-3.11.0.ebuild:54]: Seems like this is an old dep that is no longer required. Dropped.

File: media-sound/supercollider/supercollider-3.11.0.ebuild (hunk 2 above)
Comment (marcan) [reply media-sound/supercollider/supercollider-3.11.0.ebuild:59]: Looks like upstream has a spurious dep on this (both documented and in their build system). It also comes in transitively via QtWebengine, but that's optional now. I'm exposing the webengine USE flag to turn that off.

I think upstream also has a bunch of other spurious Qt deps. Let me poke them with a PR and see if I can get rid of most of them in one swoop.

**Comment by marcan** [comment]
I have a PR open with upstream to clean up the dependencies, but it looks like it might take them a while to be convinced that I didn't break anything (pretty sure I didn't) :-)

I think we should move forward here in the meantime, as this package is broken in its current state. Do you want me to pull in my PR patch and drop the deps, or leave them as is until it gets merged upstream?

**Comment by a17r** [comment]
Yes please add your patch, in ebuild please add a comment next to it referencing upstream PR.

**Comment by simonvanderveldt** [comment]
Just ran a test with this version, emerge went fine and it runs without issues as well :)
I'm so glad I can finally get rid of qtwebengine.

**Pull request closed without merging**